ta/os_test/os_test.c

```diff
@@ -409,23 +409,65 @@ static TEE_Result test_mem_access_right(uint32_t param_types,
 	TEE_Param l_params[4] = { { {0} } };
 	uint8_t buf[32];
 	TEE_TASessionHandle sess = TEE_HANDLE_NULL;
+	TEE_UUID *uuid;
 
 	if (param_types !=
 	    TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT, 0, 0, 0))
 		return TEE_ERROR_GENERIC;
-	res =
-	    TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ |
-					TEE_MEMORY_ACCESS_ANY_OWNER,
-					params[0].memref.buffer,
-					params[0].memref.size);
+
+	/* test access rights on memref parameter */
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ |
+					  TEE_MEMORY_ACCESS_ANY_OWNER,
+					  params[0].memref.buffer,
+					  params[0].memref.size);
 	if (res != TEE_SUCCESS)
 		return res;
+
 	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ,
 					  params[0].memref.buffer,
 					  params[0].memref.size);
 	if (res != TEE_ERROR_ACCESS_DENIED)
 		return TEE_ERROR_GENERIC;
 
+	/* test access rights on private read-only and read-write memory */
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ,
+					  (void *)&test_uuid, sizeof(test_uuid));
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_WRITE,
+					  (void *)&test_uuid, sizeof(test_uuid));
+	if (res == TEE_SUCCESS)
+		return TEE_ERROR_GENERIC;
+
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ |
+					  TEE_MEMORY_ACCESS_WRITE,
+					  (void *)&ret_orig, sizeof(ret_orig));
+	if (res != TEE_SUCCESS)
+		return res;
+
+	uuid = TEE_Malloc(sizeof(*uuid), 0);
+	if (!uuid)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ |
+					  TEE_MEMORY_ACCESS_WRITE,
+					  (void *)uuid, sizeof(*uuid));
+	TEE_Free(uuid);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	/* test access rights on invalid memory (at least lower 256kB) */
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ,
+					  NULL, 1);
+	if (res == TEE_SUCCESS)
+		return TEE_ERROR_GENERIC;
+
+	res = TEE_CheckMemoryAccessRights(TEE_MEMORY_ACCESS_READ,
+					  (void*)(256 * 1024), 1);
```

Comment:
From gcc man page:
```
       -Wno-int-to-pointer-cast
           Suppress warnings from casts to pointer type of an integer of a
           different size. In C++, casting to a pointer type of smaller size
           is an error. Wint-to-pointer-cast is enabled by default.
```

It's no big deal, let's wait until we get the warning and fix it then.